common/basic.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import (
+	"context"
+	"encoding/base64"
+)
+
+// NewBasicAuthHTTPClient returns a new http client, with automatic Basic authentication.
+// Specifically this means that the client will automatically add the Basic auth header
+// to every request. The username and password are provided as arguments.
+func NewBasicAuthHTTPClient( //nolint:ireturn
+	ctx context.Context,
+	user, pass string,
+	opts ...HeaderAuthClientOption,
+) (AuthenticatedHTTPClient, error) {
+	return NewHeaderAuthHTTPClient(ctx, append(opts, WithHeaders(Header{
+		Key:   "Authorization",
+		Value: "Basic " + basicAuth(user, pass),
+	}))...)
+}
+
+// shamelessly stolen from https://pkg.go.dev/net/http#Request.SetBasicAuth
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+
+	return base64.StdEncoding.EncodeToString([]byte(auth))
```

Comment:
Can we turn this into an option? It is possible that we see nonstandard requirements from some providers on how they expect this to be.

Reply:
no, basic auth is extremely standard. It will always look like this. Even the go standard http library does it this way. And on the off-chance some provider really does have some novel way of doing this (which wouldn't technically be basic auth anymore), we can just construct a custom header-client with the custom-computed header values.